nodes/villager/actions/scripts/Flee.gd

```diff
@@ -1,25 +1,72 @@
 extends "res://nodes/villager/actions/scripts/Action.gd"
 
+class_name Flee
+
 const FLEE_SOUNDS = [AudioManager.SoundType.ARGH, AudioManager.SoundType.SCREAM]
 
+var current_state = Null.new()
+
+class Null:
+	var flee: Flee
+
+	func process(delta):
+		return self
+
+	func get_label():
+		return ""
+
+class Freeze:
+	var freeze_time = 0.5
+	var flee: Flee
+
+	func _init(in_flee: Flee):
+		self.flee = in_flee
+		flee.get_animation_player().play("idle")
+
+	func process(delta: float):
+		freeze_time -= max(delta, 0)
+		if freeze_time <= 0:
+			return RunAway.new(flee)
+		return self
+
+	func get_label():
+		return "freeze"
+
+class RunAway:
+	var flee: Flee
+	
+	func _init(in_flee: Flee):
+		self.flee = in_flee
+		flee.get_animation_player().play("walk")
+		AudioManager.play_sound(FLEE_SOUNDS[randi() % FLEE_SOUNDS.size()], flee.villager)
```

Comment:
Could we use AudioStreamPlayer2D for this? Then we'll get the stereo effect based on which side of the screen they're on.

https://docs.godotengine.org/en/stable/classes/class_audiostreamplayer2d.html

Reply:
yea we could, using audio manager it will alert the "listeners" to the position of the passed in node (in this case flee.villager)

we could refactor AudioManager to use an AudioStreamPlayer2d though, i can take a look at that

Reply:
this is how i'm making villagers react to woofs and other villagers fleeing

Reply:
c90ee35